src/App.tsx

```diff
@@ -1,24 +1,246 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+  Suspense,
+  useCallback,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import 'bulma/css/bulma.css';
+import { Todo } from './types/Todo';
+// eslint-disable-next-line object-curly-newline
+import { addTodo, deleteTodo, getTodos, patchTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { USER_ID } from './constants/userid';
+import { FILTERS } from './constants/filters';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Footer } from './components/Footer';
+import { FetchContext } from './context/FetchContext';
+import { FooterContext } from './context/FooterContext';
+import { Header } from './components/Header';
+import { HeaderContext } from './context/HeaderContext';
+import { LanguageSelection } from './components/LanguageSelection';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>('');
+  const [toggleStatus, setToggleStatus] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeletingCompleted, setIsDeletingCompleted] = useState(false);
+  const [isUpdatingAllTodo, setIsUpdatingAllTodo] = useState(false);
+  const [activeFilter, setActiveFilter] = useState<FILTERS>(FILTERS.ALL);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    setToggleStatus(todos.every(todo => todo.completed));
+  }, [todos]);
+
+  const calculateNotCompletedTodo = () => {
+    return todos.filter(todo => !todo.completed).length;
+  };
+
+  const notCompletedTodoCount = useMemo(
+    calculateNotCompletedTodo,
+    [todos],
+  );
+
+  const isCompletedExist = todos.length !== notCompletedTodoCount;
+
+  const visibleTodos = useMemo(() => {
```

Comment:
useMemo() is missing its dependency array, which should include 'todos' and 'activeFilter'.

Reply:
The same as previous. Maybe there is problem with my commits, I will push it again because I have it on local version
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/87998640/83902202-f736-4f22-906a-7af8f1709ec6)